src/vala-previewer/previewer.vala

```diff
@@ -0,0 +1,111 @@
+namespace Workbench {
+
+    [DBus (name="re.sonny.Workbench.vala_previewer")]
+    public class Previewer : Object {
+
+        public void update_ui (string content, string target_id) {
+            this.builder = new Gtk.Builder.from_string (content, content.length);
+            var target = this.builder.get_object (target_id) as Gtk.Widget;
+            if (target == null) {
+                stderr.printf (@"Widget with target_id='$target_id' could not be found.\n");
+                return;
+            }
+            this.window.child = target;
+        }
```

Comment:
this doesn't support previewing GtkRoot object

if you prefer we can take care of it after this PR but please make sure you will take care of it or help out :pray:

Reply:
A problem I see is that AdwWIndow can set the content only .content, and will crash otherwise.

Reply:
When you get a GtkWindow to preview - you need to use it directly